generator/src/main/java/com/scottlogic/deg/generator/reducer/ConstraintReducer.java

```diff
@@ -67,23 +64,27 @@ public ConstraintReducer(
     }
 
     public Optional<FieldSpec> reduceConstraintsToFieldSpec(Iterable<AtomicConstraint> rootConstraints,
-                                                            Iterable<AtomicConstraint> decisionConstraints) {
-        if (rootConstraints == null || !rootConstraints.iterator().hasNext()) {
-            return Optional.of(FieldSpec.Empty);
-        }
+                                                            Set<FieldSpec> decisionConstraints) {
 
-        final FieldSpec decisionConstaintsFieldSpec = fieldSpecFactory.toMustContainRestrictionFieldSpec(
+        Optional<FieldSpec> rootFieldSpec =
+            rootConstraints == null ? Optional.of(FieldSpec.Empty)
```

Comment:
Minor point, line spacing here can make the ternary easier to read

```suggestion
            rootConstraints == null 
                ? Optional.of(FieldSpec.Empty)
```